src/OpenTelemetry/Logs/LogRecordData.cs

```diff
@@ -0,0 +1,130 @@
+// <copyright file="LogRecordData.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+#nullable enable
+
+using System;
+using System.Diagnostics;
+using Microsoft.Extensions.Logging;
+
+namespace OpenTelemetry.Logs
+{
+    /// <summary>
+    /// Stores details about a log message.
+    /// </summary>
+    internal struct LogRecordData
+    {
+        internal DateTime TimestampBacking = DateTime.UtcNow;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="LogRecordData"/> struct.
+        /// </summary>
+        /// <remarks>
+        /// Note: The <see cref="Timestamp"/> property is initialized to <see
+        /// cref="DateTime.UtcNow"/> automatically.
+        /// </remarks>
+        /// <param name="activity">Optional <see cref="Activity"/> used to populate context fields.</param>
+        public LogRecordData(Activity? activity = null)
+        {
+            if (activity != null)
+            {
+                this.TraceId = activity.TraceId;
+                this.SpanId = activity.SpanId;
+                this.TraceState = activity.TraceStateString;
+                this.TraceFlags = activity.ActivityTraceFlags;
+            }
+            else
+            {
+                this.TraceId = default;
+                this.SpanId = default;
+                this.TraceState = null;
+                this.TraceFlags = ActivityTraceFlags.None;
+            }
+        }
+
+        /// <summary>
+        /// Gets or sets the log timestamp.
```

Comment:
nit: consider a comment here that the value is stored as UTC.

Reply:
Updated!